internal/server/data/access_key.go

```diff
@@ -297,19 +295,47 @@ func ValidateRequestAccessKey(tx *Transaction, authnKey string) (*models.AccessK
 		if now.After(t.InactivityTimeout) {
 			return nil, ErrAccessInactivityTimeout
 		}
-
-		origTimeout := t.InactivityTimeout
 		t.InactivityTimeout = now.Add(t.InactivityExtension)
-		// Throttle updates when the key is used frequently. Uses the
-		// same value as server.lastSeenUpdateThreshold.
-		if t.InactivityTimeout.Sub(origTimeout) > 2*time.Second {
-			if err := UpdateAccessKey(tx, t); err != nil {
-				return nil, err
-			}
-		}
 	}
 
-	return t, nil
+	err = updateAccessKeyLastSeenAt(tx, t)
+	return t, err
+}
+
+// lastSeenUpdateThreshold is the duration of time that must pass before a
+// LastSeenAt value for a user, access key, or destination is updated again.
+// This prevents excessive writes when a single user performs many requests
+// in a short period of time.
+const lastSeenUpdateThreshold = 2 * time.Second
```

Comment:
Do we need to throttle this if we're skipping locked rows below anyways?

Reply:
Good question. I think each of these has a different purpose. The throttling helps reduce the total number of DB queries, where as the skip locked is to prevent queries from blocking each other. If the API requests are spaced out by 200ms or so then the skip locked doesn't do anything, but throttling will still help us skip a good number of database updates.

The throttling is an optimization, so we don't absolutely need it, but reducing the number of updates is still really nice to keep load off the database. Since it's only 3 lines I'd still like to keep it.